agent/src/util/device_plugin_service.rs

```diff
@@ -899,6 +921,24 @@ mod device_plugin_service_tests {
         );
     }
 
+    // Test that a Device and Configuration's properties are aggregated and that
+    // a Device property overwrites a Configuration's.
+    #[test]
+    fn test_get_all_broker_properties() {
+        let mut device_properties = HashMap::new();
+        device_properties.insert("ENDPOINT".to_string(), "123".to_string());
+        device_properties.insert("OVERWRITE".to_string(), "222".to_string());
+        let mut configuration_properties = HashMap::new();
+        configuration_properties.insert("USE HD".to_string(), "true".to_string());
+        configuration_properties.insert("OVERWRITE".to_string(), "111".to_string());
+        let all_properties =
```

Comment:
this is a great test ... i wasn't sure what we wanted to happen in this case and it is good to have a test to point to and say, "this is what we expect"